clayui.com/src/components/Editor/index.js

```diff
@@ -38,39 +36,31 @@ const Editor = ({
 	preview = true,
 	scope = {},
 }) => {
-	let defaultSnippet = {
-		code: '',
-		disabled: false,
-		format: true,
-		imports,
-		name: 'React',
-	};
-
 	if (Array.isArray(code)) {
-		const codeToFormat = [...code];
-
-		codeToFormat.forEach((snippet) => {
-			if (snippet.format) {
-				snippet.code = formatCode(snippet.code);
-			}
+		code = code.map(({code, format = true, ...other}) => {
+			return {
+				...other,
+				code: format ? formatCode(code) : code,
+			};
 		});
-
-		code = codeToFormat;
-
-		defaultSnippet = code[REACT_SNIPPET_INDEX];
 	} else {
-		code = formatCode(code);
-
-		defaultSnippet.code = code;
+		code = [
+			{
+				disabled: false,
+				imports,
+				name: 'React',
+				value: formatCode(code),
+			},
+		];
 	}
 
 	const [collapseCode, setCollapseCode] = useState(false);
-	const [activeTabKeyValue, setActiveTabKeyValue] = React.useState(0);
-	const [values, setValues] = React.useState(code);
+	const [activeIndex, setActiveIndex] = React.useState(0);
+	const [snippets, setSnippets] = React.useState(code);
 
 	return (
 		<LiveProvider
-			code={defaultSnippet.code}
+			code={snippets[0].value}
```

Comment:
no need for us to use the name `default` since we really just want to default to the first index